docs/spec.md

```diff
@@ -709,31 +709,32 @@ Afterwards, within each `process_group`:
 
 #### Inputs
 
-| Label | Name                    | Type                                         | Constraints |
-|-------|-------------------------|----------------------------------------------|-------------|
-| (I1)  | `operand`               | tensor or per-tensor quantized tensor        | (C1), (C6)  |
-| (I2)  | `all_gather_dim`        | constant of type `si64`                      | (C1), (C6)  |
-| (I3)  | `replica_groups`        | 2-dimensional tensor constant of type `si64` | (C2-C4)     |
-| (I4)  | `channel_id`            | constant of type `si64`                      | (C5)        |
-| (I5)  | `use_global_device_ids` | constant of type `i1`                        | (C5)        |
+| Label | Name                    | Type                                         | Constraints      |
+|-------|-------------------------|----------------------------------------------|------------------|
+| (I1)  | `operand`               | tensor or per-tensor quantized tensor        | (C1), (C2), (C7) |
+| (I2)  | `all_gather_dim`        | constant of type `si64`                      | (C1), (C2), (C7) |
+| (I3)  | `replica_groups`        | 2-dimensional tensor constant of type `si64` | (C3-C5)          |
+| (I4)  | `channel_id`            | constant of type `si64`                      | (C6)             |
+| (I5)  | `use_global_device_ids` | constant of type `i1`                        | (C6)             |
 
 #### Outputs
 
 | Name     | Type                                  | Constraints |
 |----------|---------------------------------------|-------------|
-| `result` | tensor or per-tensor quantized tensor | (C6)        |
+| `result` | tensor or per-tensor quantized tensor | (C7)        |
 
 #### Constraints
 
-* (C1) `0 <= all_gather_dim < rank(operand)`.
-* (C2) `is_unique(replica_groups)`.
-* (C3) `size(replica_groups)` is defined as:
+* (C1) `dim(operand, all_gather_dim) > 0`.
```

Comment:
I'm not sure where this constraint is coming from. Perhaps let's leave open a Specification Compliance ticket and leave this to another PR? Everything else LGTM.

Reply:
Created #1745